test/HostActivationTests/GivenThatICareAboutDotnetTestXunitScenarios.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+using Microsoft.DotNet.CoreSetup.Test;
+using System.Runtime.InteropServices;
+using System.IO;
+
+namespace Microsoft.DotNet.Tools.Publish.Tests
+{
+    public class GivenThatICareAboutDotnetTestXunitScenarios
+    {
+        private string DotnetTestXunitVersion => "1.0.0-rc2-192208-24";
+        private RepoDirectoriesProvider RepoDirectories { get; set; }
+
+        public GivenThatICareAboutDotnetTestXunitScenarios()
+        {
+            RepoDirectories = new RepoDirectoriesProvider();
+        }
+
+        [Fact]
+        public void Muxer_activation_of_dotnet_test_XUnit_on_Portable_Test_App_Succeeds()
+        {
+            var portableTestAppFixture = new TestProjectFixture("PortableTestApp", RepoDirectories)
+                .EnsureRestored(RepoDirectories.CorehostPackages, RepoDirectories.CorehostDummyPackages)
+                .BuildProject();
+
+            ActivateDotnetTestXunitOnTestProject(RepoDirectories, portableTestAppFixture);
+        }
+
+        [Fact]
+        public void Muxer_activation_of_dotnet_test_XUnit_on_Standalone_Test_App_Succeeds()
+        {
+            if ( ! RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                Console.WriteLine("Skipped Test: https://github.com/dotnet/core-setup/issues/121");
+                return;
+            }
+
+            var standaloneTestAppFixture = new TestProjectFixture("StandaloneTestApp", RepoDirectories);
+            standaloneTestAppFixture
+                .EnsureRestored(RepoDirectories.CorehostPackages, RepoDirectories.CorehostDummyPackages)
+                .BuildProject(runtime: standaloneTestAppFixture.CurrentRid);
+
+            ActivateDotnetTestXunitOnTestProject(RepoDirectories, standaloneTestAppFixture);
+        }
+
+        public void ActivateDotnetTestXunitOnTestProject(
+            RepoDirectoriesProvider repoDirectories, 
+            TestProjectFixture testProjectFixture)
+        {
+            var dotnet = testProjectFixture.BuiltDotnet;
+
+            var dotnetTestXunitDll = FindDotnetTestXunitDll(repoDirectories, DotnetTestXunitVersion);
+            var depsJson = testProjectFixture.TestProject.DepsJson;
+            var runtimeConfig = testProjectFixture.TestProject.RuntimeConfigJson;
+            var additionalProbingPath = RepoDirectories.NugetPackages;
+            var appDll = testProjectFixture.TestProject.AppDll;
+
+            dotnet.Exec(
```

Comment:
Yeah we can do some parsing of the output. That's a good idea.

---
In reply to: [66346333](https://github.com/dotnet/core-setup/pull/97#discussion_r66346333) [](ancestors = 66346333)